test/functional/ui/screen_advanced_spec.lua

```diff
@@ -0,0 +1,79 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear, feed, insert = helpers.clear, helpers.feed, helpers.insert
+local execute = helpers.execute
+
+
+describe('Screen rendering', function()
+  local screen
+  local colors = Screen.colors
+  local hl_colors = {
+    NonText = colors.Blue,
+    Search = colors.Yellow,
+    Message = colors.Red,
+  }
+
+  before_each(function()
+    clear()
+    screen = Screen.new(40, 7)
+    screen:attach()
+    --ignore highligting of ~-lines
+    screen:set_default_attr_ids( {
+      [1] = {foreground = Screen.colors.Brown},
+      [2] = {bold = true, foreground = Screen.colors.Brown},
+      [3] = {foreground = hl_colors.Message},
+    })
+    screen:set_default_attr_ignore( {{bold=true, foreground=hl_colors.NonText}} )
```

Comment:
Also I would suggest removing this, partly because you will be touching code that controls `NonText` highlighting most likely (and partly because I also think it was a mistake to keep this after introducing `snapshot_util()` , it is better tests assert the screen state exactly than they look a bit better). Anyway, as your existing tests has no NonText highlighting, there would be no extra work beyond removing this line and then use `snapshot_util()` as usual.